src/ape/api/networks.py

```diff
@@ -610,6 +611,65 @@ def get_python_types(  # type: ignore[empty-body]
             Union[Type, Sequence]: The Python types for the given ABI type.
         """
 
+    def decode_custom_error(
+        self,
+        data: HexBytes,
+        address: AddressType,
+        **kwargs,
+    ) -> Optional[CustomError]:
+        """
+        Decode a custom error class from an ABI defined in a contract.
+
+        Args:
+            data (HexBytes): The error data contining the selector
+              and input data.
+            address (AddressType): The address of the contract containing
+              the error.
+            **kwargs: Additional init kwargs for the custom error class.
+
+        Returns:
+            Optional[CustomError]: If it able to decode one, else ``None``.
+        """
+
+        # Use an instance (required for proper error caching).
+        contract = self.chain_manager.contracts.instance_at(address)
+
+        selector = data[:4]
```

Comment:
Probably specific to ethereum, but fine for now until someone else needs a different mechanism to handle it (just add a comment)

Reply:
I was thinking of implementing this there

Reply:
Don't think it's a huge deal now, probably all EVM networks handle it this way, but a note to come back and refactor into EcosystemAPI would be nice

Reply:
i just went ahead and moved it! wasnt too bad